src/Hooks/useSettings.jsx

```diff
@@ -0,0 +1,55 @@
+import React, { createContext } from 'react';
+import PropTypes from 'prop-types';
+
+export const SettingsContext = createContext(null);
+const { body } = document;
+
+export class SettingsProvider extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      theme: {
+        isShowSettings: false,
+        toggleSettings: this.toggleSettings,
+        setLightTheme: this.toggleTheme.bind(this, 'light'),
+        setDarkTheme: this.toggleTheme.bind(this, 'dark'),
+        setDarkDimmedTheme: this.toggleTheme.bind(this, 'dark-dimmed'),
+      },
```

Comment:
you violate several principles here, because these all fields are not related to theme object, I'd recommend to remove theme object at all and just pass a state with all fields directly

Reply:
```
    this.state = {
      isShowSettings: false,
      toggleSettings: this.toggleSettings,
      setLightTheme: this.toggleTheme.bind(this, 'light'),
      setDarkTheme: this.toggleTheme.bind(this, 'dark'),
      setDarkDimmedTheme: this.toggleTheme.bind(this, 'dark-dimmed'),
    };
```
Fixed.
I don't remember why I couldn't do it before. Thanks.